### PR TITLE
cmd/snap, daemon: make the connectivity check use GET

### PR DESCRIPTION
**Pull request by chipaca**
With this, all non-hidden debug commands can run unauthorized.

### CODE REVIEW

File: daemon/api_debug.go [hunk 1]
```diff
@@ -63,18 +63,43 @@ func getBaseDeclaration(st *state.State) Response {
 	}, nil)
 }
 
+func checkConnectivity(theStore snapstate.StoreService) Response {
```
Comment (zyga) [inline daemon/api_debug.go]: This looks good.

File: daemon/api_debug.go [hunk 2]
```diff
@@ -63,18 +63,43 @@ func getBaseDeclaration(st *state.State) Response {
...
+}
+
 func getDebug(c *Command, r *http.Request, user *auth.UserState) Response {
 	query := r.URL.Query()
 	aspect := query.Get("aspect")
-	if aspect != "base-declaration" {
+	st := c.d.overlord.State()
+	switch aspect {
+	case "base-declaration":
+		st.Lock()
```
Comment (zyga) [inline daemon/api_debug.go]: Perhaps consider moving locking ahead of the switch statement. We can move it inside the methods if there is a need for it.

File: daemon/api_debug_test.go [hunk 3]
```diff
@@ -90,19 +98,34 @@ func (s *postDebugSuite) TestPostDebugConnectivityHappy(c *check.C) {
...
+	_ = s.daemon(c)
 
 	s.connectivityResult = map[string]bool{
 		"good.host.com": true,
 		"bad.host.com":  false,
 	}
 
-	rsp := postDebug(debugCmd, req, nil).(*resp)
+	var rsp *resp
+	if post {
```
Comment (zyga) [inline daemon/api_debug_test.go]: Tests look fine. Thank you for handling both cases.

File: daemon/api_debug.go (hunk 1 above)
Comment (pedronis) [reply daemon/api_debug.go]: looking a bit more it's a bit unclear why this is not taking the state and getting the store on its own as well

**Comment by pedronis** [comment]
I pushed a slight reorg with checkConnectivity taking state and a Store and changed locking accordingly.

File: daemon/api_debug.go [hunk 4]
```diff
@@ -103,23 +126,7 @@ func postDebug(c *Command, r *http.Request, user *auth.UserState) Response {
...
-		for host, reachable := range checkResult {
-			if !reachable {
-				status.Connectivity = false
-				status.Unreachable = append(status.Unreachable, host)
-			}
-		}
-		sort.Strings(status.Unreachable)
-
-		return SyncResponse(status, nil)
+		return checkConnectivity(st)
```
Comment (chipaca) [inline daemon/api_debug.go]: this will panic

Comment (chipaca) [reply daemon/api_debug.go]: no it won't

**Pull request merged**